DPGAnalysis/SiStripTools/plugins/APVCyclePhaseProducerFromL1ABC.cc

```diff
@@ -186,30 +189,54 @@ void APVCyclePhaseProducerFromL1ABC::produce(edm::Event& iEvent, const edm::Even
   if (iEvent.run() >= _firstgoodrun) {
     Handle<L1AcceptBunchCrossingCollection> pIn;
     iEvent.getByToken(_l1abccollectionToken, pIn);
-
+    Handle<TCDSRecord> tcds_pIn;
+    iEvent.getByToken(_tcdsRecordToken, tcds_pIn);
+    const auto& tcdsRecord = *tcds_pIn.product();
+    bool useTCDS(tcds_pIn.product());
```

Comment:
Is the intention to use `useTCDS` to check whether a `TCDSRecord` exists or not?

Reply:
Yes. I could not find a better way; if you can suggest one it would be awesome!

About the clang-format, the point is that it seems it checks the rules only for the changes, and does not allow me to fix similarly wrong syntax that was already there in the module, which does not make sense to me.

Reply:
> Yes. I could not find a better way; if you can suggest one it would be awesome!

```
if(tcds_pIn.isValid()) {
   // TCDSRecord exists
} else {
  // TCDSRecord does not exist
}
```
The `tcds_pIn.product()` throws an exception if the product does not exist.

> About the clang-format, the point is that it seems it checks the rules only for the changes, and does not allow me to fix similarly wrong syntax that was already there in the module, which does not make sense to me.

Our format definition imposes maximum width for the line, and long lines get broken to multiple lines in some way. Therefore adding one character to a line can lead to reformatting of the line.

Reply:
Thank you very much!